website/more_documentation/button_documentation.py

```diff
@@ -33,3 +38,21 @@ async def await_button_click() -> None:
             ui.label('Two')
             await b.clicked()
             ui.label('Three')
+
+    @text_demo('Disable button with a context manager', '''
+        This showcases a async context manager that can be used to disable a button for the duration of an async process.
+    ''')
+    async def disable_context_manager() -> None:
+        @asynccontextmanager
+        async def disable(element: DisableableElement) -> None:
+            element.disable()
+            try:
+                yield
+            finally:
+                element.enable()
+
+        async def disable_and_sleep_3(button: Button) -> None:
+            async with disable(button):
+                await sleep(3)
+
+        b = ui.button("Disable for 3 seconds", on_click=lambda: disable_and_sleep_3(b))
```

Comment:
We could also write `lambda e: disable_and_sleep_3(e.sender)`, avoiding the need for the local variable `b`.